Esri/arcgis-runtime-samples-ios
# Update: NMEA sample with GNSS device API

## Goals

To showcase the new NMEA LDS constructor in U11 that connect to a real device easily without pushing NMEA data manually.

## Changes and Updates

- talked to MFi device manufacturers to whitelist the samples app for next release, so that Apple can approve Samples app to connect to external accessories.
- update source code to support both mock data and U11 constructor, and to update UI and logics
- update README

## Demo

Please see the internally shared video - don't post here to avoid leaking my accurate location info 😅 

## How to Test

- If you don't have a GNSS surveyor, simply test the mock data functionalities, and look for any logic bugs in the code.
- If you have a GNSS surveyor that is one of these 3 brands: Bad Elf, EOS, SXBlue, then try to connect the device and switch back and forth to see if the sample works as expected

## Todolist

- [ ] after the changes are approved, have Divesh take a look
- [ ] verify the sample with 3 GNSS devices that are currently possessed by different teams
    - [x] Ting: BadElf
    - ~~Nick: EOS~~ Nick has sent back the GNSS accessory. 😢 `cocoa/issues/10553#issuecomment-2824557`
    - [ ] Apps?: SXBlue
- [ ] before merging PR, will add an iOS specific design doc to internal repo
- [ ] before merging PR, follow up with manufacturer representative to see if we get our bundle id whitelisted

## Extra Information

The BadElf CTO Brett offers to help to review the verbiage for adding GNSS receiver support to an app. So after the code is good and people are happy with the README, we can send him a link to this PR to get some insights!
